From 2256b28be7d2dad7c93247ef5e761fd5acc1025f Mon Sep 17 00:00:00 2001
Message-ID: <2256b28be7d2dad7c93247ef5e761fd5acc1025f.1719691458.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1719691458.git.sharlatanus@gmail.com>
References: <cover.1719691458.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 24 Jun 2024 21:24:47 +0100
Subject: [PATCH 21/26] gnu: Add go-atomicgo-dev-schedule.

* gnu/packages/golang-xyz.scm (go-atomicgo-dev-schedule): New variable.

Change-Id: Iee83258d43320bbd99170c9bcb21f7d6984366bc
---
 gnu/packages/golang-xyz.scm | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 4bdcc05c47..f836449d54 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -111,6 +111,31 @@ (define-public go-atomicgo-dev-cursor
 any kind.")
     (license license:expat)))
 
+(define-public go-atomicgo-dev-schedule
+  (package
+    (name "go-atomicgo-dev-schedule")
+    (version "0.1.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/atomicgo/schedule")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "13zrmf9jagqjvjjckyqlvr889y2gxf22iz42l6j2zmgy9klbn6vl"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:go go-1.18
+      #:import-path "atomicgo.dev/schedule"))
+    (home-page "https://atomicgo.dev/schedule")
+    (synopsis "Easily schedule non-blocking tasks in Golang")
+    (description
+     "This package provides a simple scheduler which, can run a function at a
+given time, in a given duration, or repeatedly at a given interval.")
+    (license license:expat)))
+
 (define-public go-bazil-org-fuse
   (package
     (name "go-bazil-org-fuse")
-- 
2.41.0

